app/services/asset_manager/attachment_updater/access_limited_updates.rb

```diff
@@ -5,15 +5,28 @@ def self.call(attachment_data)
       access_limited = AssetManagerAccessLimitation.for(attachment_data.access_limited_object)
     end
 
-    Enumerator.new do |enum|
-      enum.yield AssetManager::AttachmentUpdater::Update.new(
-        attachment_data, attachment_data.file, access_limited:
-      )
-
-      if attachment_data.pdf?
+    if attachment_data.assets.empty?
+      Enumerator.new do |enum|
         enum.yield AssetManager::AttachmentUpdater::Update.new(
-          attachment_data, attachment_data.file.thumbnail, access_limited:
+          nil, attachment_data, attachment_data.file, access_limited:
         )
+
+        if attachment_data.pdf?
+          enum.yield AssetManager::AttachmentUpdater::Update.new(
+            nil, attachment_data, attachment_data.file.thumbnail, access_limited:
+          )
+        end
+      end
+    else
+      # Both the file and its thumbnail are now represented by an Asset
```

Comment:
In all these files `xxx_updates` we have kept the two possible flows coexisting and the new one should only run if assets are present.